src/openjd/model/_step_param_space_iter.py

```diff
@@ -460,3 +745,128 @@ def __getitem__(self, index: int) -> TaskParameterSet:
 
     def iter(self) -> RangeExpressionIdentifierNodeIterator:
         return RangeExpressionIdentifierNodeIterator(self)
+
+
+class AdaptiveContiguousChunkIdentifierNodeIterator(NodeIterator):
+    """Iterator for an AdaptiveChunkIdentifierNode
+
+    Attributes:
+        _index: The index into _node.range for the start of the next chunk.
+        _node: The AdaptiveChunkIdentifierNode this is iterating over.
+    """
+
+    _index: int
+    _node: AdaptiveContiguousChunkIdentifierNode
+
+    def __init__(self, node: AdaptiveContiguousChunkIdentifierNode):
+        self._node = node
+        self.reset_iter()
+
+    def reset_iter(self) -> None:
+        self._index = 0
+
+    def next(self, result: TaskParameterSet) -> None:
+        if self._index >= len(self._node.range):
+            raise StopIteration()
+
+        chunks_default_task_count = self._node.chunks_default_task_count
+        range = self._node.range
```

Comment:
nit: this shadows the built-in `range` function. Can we rename it to `parameter_range`?

Reply:
Will do.